### ISSUE TITLE
Coleta de dados para as métricas da He4rt

### ISSUE THREAD

**Issue by yayaflc** [he4rt/he4rt-bot-next#39]
Conforme o crescimento da He4rt e sua expansão de eventos/projetos, queremos que a qualidade siga o mesmo ritmo, trazendo uma comunidade mais igualitária e unida.

A sugestão seria coletar dados importantes pelo `/apresentar` para criação de métricas e infográficos da He4rt, como:
- Cidade;
- Estado;
- País;
- Gênero;
-  Orientação Sexual,
- Etnia.

> Esses dados seriam exclusivos da Administração e utilizados de maneira anônima, seguindo protocolos LGPD.

Queremos ter noção de onde nossa comunidade é mais presente, para assim levar a He4rt Meetup e outros projetos para esses lugares. Além disso, possibilitaria consultar o tamanho do desfalque de cada minoria, assim seremos capazes de nos dedicar em projetos que incentivem mais a presença de cada grupo minoritário.

1. Você se sentiria confortável em compartilhar seus dados? De qual maneira podemos abordar? 
3. Qual a forma mais segura e não-invasiva para a coleta dessas métricas?
4. Você acredita que todas as perguntas devem ser obrigatórias? (ex. apenas Estado, Gênero)
5. Teria algum outro dado importante para podermos metrificar nosso crescimento?

**Comment by JVictorV** [comment]
1 - Por mim seria tranquilo, acho que um fluxo dentro do bot ja seria suficiente
2 - Acho que deixar esses dados num banco e seguir as normas da LGPD ja e o suficiente, a coleta pode ser opicional tmb
3 - Nenhuma pergunta deveria ser obrigatoria
4 - Mes de inicio como dev (pra saber tempo de carreira), se ja trabalha ou nao e nivel de senioridade

**Comment by heltonp21** [comment]
1. Sim, é importante pra comunidade.
2. Acredito que o Docs possa ser um bom tipo de coleta.
3. Quem se sentir confortável que responda, não deveria ser todas obrigatórias.
4. Retenção de usuários.

**Comment by henri1i** [comment]
1. Sim. Não vejo muito problema na abordagem atual.
2. Acho que falta a confirmação de que esses dados não serão comercializados e etc, já que isso pode acabar sendo uma preocupação.
3. Não, pode causar algum tipo de desconforto desnecessário na primeira interação do membro com a comunidade. Acho isso um pouco perigoso.
4. Não consigo pensar em nada agora.

Obs: Fico com receio de adicionar muitas perguntas ao comando de apresentação. Tenho medo do novo membro desistir de responder no meio, já que essa etapa pode acabar sendo um pouco demorada.

**Comment by Canhassi12** [comment]
1- Sim, sem problemas, mas serem coisas opcionais colocar ou não.
2- Acho que justamente ser opcional para qualquer um.
3- == 
4- Uma sugestão que eu tenho, que serveria para envolver mais outros membros da comunidade, é propor alguns projetos e meio que fazer desafios samanais ou mensais(A se pensar). E pegar todos os membros que estão dispostos, montar pequenos grupos dependendo da quantidade de gente e suas tecnologias. E em determinado tempo, na reunião ter uma avaliação dos projetos, ter algumas referencias backend, frontend e debater de acordo com o que foi aprensentado com a galera. É mais uma forma de interação e integrar iniciantes na área a fazer projetos em grupo, pode ser algo importante para conhecer o mercado.

**Comment by kelvinwieth** [comment]
1. Depende. Existem garantias de que esses dados estarão seguros? Quem vai ter acesso ao banco?
2. Bot chamando na DM do Discord tá de bom tamanho. O próprio Discord já encripta as mensagens.
3. Acredito que todas devem ser opcionais.
4. Acho que seria bom medir a xp profissional dos membros. Com isso daria pra direcionar melhor as palestras, cursos etc.

**Comment by PedroPMS** [comment]
> Minha opinião acredito que Estado, País e Gênero deveriam ser obrigatórios.
> 
> * Estado/País: são dados muito amplos, sem expor endereço mais sensível, que servem para podermos localizar nossos membros pelo Brasil inteirinho (e fora dele), assim podemos saber aonde podemos trazer mais eventos presenciais.
> * Gênero: dado importante devido ao He4rt Delas. Podemos ter um acesso melhor sobre pessoas que se identificam como mulheres e pessoas não-binárias, já que no `/apresentar` existe casos de pessoas pegando acesso ao #delas por engano. Sem falar que podemos coletar a métrica de sucesso do He4rt Delas, conforme passa os meses e entram mais mulheres.
> 
> Também penso que se deixa tudo sem ser obrigatório, muitas vezes não respondem nada.
> 
> Gostei do ponto do Henri que eu não tinha notado, sobre **quantidade excessiva de perguntas** no `/apresentar`, porém a ideia do Daniel sobre dividir para outro comando abre espaço para as pessoas não darem atenção e não responderem, e assim pegariamos bem menos informações e até mesmo não iria pra frente a ideia de coletar as métricas... Precisamos pensar direitinho sobre.

O gênero pode ser obrigatório mas ter a opção de "Prefiro não informar, algo assim"

**Comment by henri1i** [comment]
Pensei em adicionar uma progress bar no comando `/apresentar`, pra informar o usuário sobre quantas perguntas faltam pra conclusão do formulário. 
Não saber se faltam 8 ou 80 perguntas pode fazer algum dos usuários desistirem no meio do processo.

**Comment by laisacsts** [comment]
1 - Me sinto confortável em compartilhar meus dados quando as informações sobre uso são claras e diretas.
2 - Formulário anônimo porque vai receber dados sensíveis (gênero). (Se for com o `/apresentar` as respostas vão ficar anônimas pro pessoal da administração?)
3 - Sim, acredito que deixar as respostas opcionais pode dar uma bagunçada nos resultados.
4 - Idade.

**Comment by henri1i** [comment]
> @laisacsts 
> 4 - Idade.

Acredito que já exista a issue pra adicionar a coleta de idade ao comando: https://github.com/he4rt/he4rt-bot-api/pull/15

**Comment by henechen** [comment]
1. Levando em consideração o background da comunidade, sim. Porém, para novos membros que nunca ouviram falar da comunidade por exemplo, talvez um informe de como a comunidade é (mostrando talvez algum projeto grande) dê mais uma segurança para a pessoa que irá se apresentar no canal. A atual maneira até o momento acredito ser ok.

2. Se possível uma maneira que seja direto pelo Discord mesmo, e que então fique armazenado em um lugar. Existem pessoas que ao encontrarem "burocracias" como abrir um link de Google Docs, etc, acaba desanimando de fazer o processo e como o objetivo é coletar estes dados para métrica, é de importância trazer possivelmente a forma mais prática possível de se fazer o mesmo.

3. Não acredito que deva ser obrigatória mas deve-se reiterar durante o processo da segurança desses dados e **principalmente** a necessidade dos mesmos para eventos e projetos futuros.

4. Especificamente não sei um dado que seja importante mas quanto mais informações forem disponibilizadas, melhor ficaria a questão da organização.

**Comment by joaocstro** [comment]
Concordo com o @henri1i e o @DanielHe4rt no que diz respeito trazer duas etapas para o processo do `/apresentar`. Também acredito que a informação da **cidade** do usuário não deveria ser coletada pelos seguintes motivos:

1. Não vejo essa informação ajudando muito na tomada de decisão da localização dos eventos. Imagino que todos vão acabar sendo feitos em uma capital;
2. É uma informação muito específica e pode trazer grandes problemas caso os dados vazem;
3. Acredito que a maioria dos usuários não se sentiriam confortáveis em compartilhar sua cidade em um servidor de Discord que acabaram de entrar.

Na minha avaliação, é uma informação muito invasiva, que não traz muitos benefícios e acaba não justificando os riscos de ser coletada.

Sobre o que a @yayaflc disse a respeito dos usuários ignorarem a segunda etapa, eu acredito que isso pode ser resolvido através de algum incentivo para que os usuários finalizem o processo. Isso pode vir no formato de uma tag exclusiva, acesso a comandos, boost no ganho de XP, etc.

**Comment by Rinyaresu** [comment]
1 - Eu acho que a abordagem atual é tranquila. Eu gosto dela.
2 - Acho que ser opcional é o melhor caminho.
3 - Nenhuma
4 - Tempo de carreira e idade.

Gosto da ideia de fazer em duas etapas que o @DanielHe4rt. Mas é importante levar em consideração que a quantidade de perguntas pode ser um **desafio** para quem está começando na comunidade.

**Comment by yayaflc** [comment]
> Concordo com o @henri1i e o @DanielHe4rt no que diz respeito trazer duas etapas para o processo do `/apresentar`. Também acredito que a informação da **cidade** do usuário não deveria ser coletada pelos seguintes motivos:
> 
> 1. Não vejo essa informação ajudando muito na tomada de decisão da localização dos eventos. Imagino que todos vão acabar sendo feitos em uma capital;

A informação da cidade seria para projetos presenciais além do He4rt Meetup (que sim, acaba sendo em capitais já que está sendo realizado em parceria com as unidades da escola 42, até então), porém existe a possibilidade de novos eventos presenciais surgirem, para dar oportunidade para aqueles que não possuem condições de viajar para a capital ou por não ter iniciativa de encontro entre membros (por falta de iniciativa). Eventos presenciais de porte pequeno para essas cidades pode ser uma realidade muito em breve.

**Comment by enesdeveloper** [comment]
1. Sim, é uma boa abordagem.
2. Uma etapa extra no apresentar, com bonificação de alguma forma em forma de XP, boost de XP, comandos exclusivos e etc.
3. Como seria uma etapa extra, de certa forma já é opcional, então seria interessante colocar alguns campos como obrigatórios como estado, país e gênero. 
4. Tempo de carreira, senioridade e idade.

**Comment by LucasCrossDimitri** [comment]
> 1. Sim. Não vejo muito problema na abordagem atual.
> 
>     2. Acho que falta a confirmação de que esses dados não serão comercializados e etc, já que isso pode acabar sendo uma preocupação.
> 
>     3. Não, pode causar algum tipo de desconforto desnecessário na primeira interação do membro com a comunidade. Acho isso um pouco perigoso.
> 
>     4. Não consigo pensar em nada agora.
> 
> 
> Obs: Fico com receio de adicionar muitas perguntas ao comando de apresentação. Tenho medo do novo membro desistir de responder no meio, já que essa etapa pode acabar sendo um pouco demorada.

como o daniel comentou, apoio a ideia do henrizão tmb!

**Comment by LucasCrossDimitri** [comment]
como o daniel comentou, apoio a ideia do henrizão tmb!

> 1. Sim. Não vejo muito problema na abordagem atual.
> ....

**Comment by caio-emidio** [comment]
Você se sentiria confortável em compartilhar seus dados? De qual maneira podemos abordar?
 1- Alguns dados nao vejo como poderiam ajudar dentro da He4rt/Comunidade tais como orientacao sexual. Talvez deixar como opcional seja claro.
Qual a forma mais segura e não-invasiva para a coleta dessas métricas?
 2- A abordagem atual eh interessante. Talvez ter algo que diga se a pessoa quer ou nao apresentar as informacoes ao final. 
Você acredita que todas as perguntas devem ser obrigatórias? (ex. apenas Estado, Gênero)
 3- Acredito que todas as respostas devem ser opcionais.
Teria algum outro dado importante para podermos metrificar nosso crescimento?
 4- Quais informacoes a he4rt usa de dados oriundos de outras plataformas (ex: dados do discord) .
 Se vale a minha vivencia de comunidades. Eu lembro que na minha ultima comunidade nos queriamos criar uma micro social network. So para alinhar esses pontos de dados. Novos projetos e etc. Bem parecido com o que a He4rt faz hoje com o 4noobs, he4rt delas, apoio aos streamers, reunioes e etc... Tudo isso linkado sempre ao discord/site e twitter. 
Um projeto desses totalmente geraria dados que poderiam ser extremamente interessantes para a comunidade e o futuro da he4rt.

**Comment by sc0v1n0** [comment]
Bom dia,
Poderia ter algo mais relacionando as dificuldades para os desenvolvedores, vejo que no discord existem muitos começando talvez pode ser analisado quais pontos podem ser trazidos para discord para ajudar.
- Na sua cidade existem empresas de T.I ?
- Na sua cidade existe faculdade de alguma área de T.I ?
- Você trabalha na área de T.I ?
    - Você trabalha presencial ?
        - Existe dificuldade para chegar ao trabalho ?
        - O ambiente de trabaho e propício  para trabalhar com T.I ?
        - Você recebeu um computador para trabalhar ?
        - Existe conexão de internet para todos do ambiente ?
        
        - Não?
           - Você utiliza o seu computador próprio ?
           - O seu computador atende as requesitos das ferramentas utilizadas para desenvolvimento ?  
    
Mais tarde adiciono mais opções.....

**Comment by Luisnadachi** [comment]
Concordo com o ponto do @henri1i, pode deixar essa parte para um outro comando para não tornar o processo do `/apresentar` cansativo para os novos membros, sobre os pontos perguntas.

- Sim, acho que pode abordar de uma maneira transparente, seguindo os pontos bases da LGPD principalmente para manter as pessoas dboa.
- Foi o que citei acima.
- Acho que em questão de região onde mora pode deixar como obrigatória, mas com relação a pessoa em si pode botar uma opção de "prefiro não informar", algo do tipo.
- Por agora não consigo pensar em nenhum.

**Comment by paulorievrs** [comment]
Acho que talvez no apresentar colocar talvez uma opção pro cara selecionar por onde ele conheceu o servidor/comunidade, já que a ideia da issue era obter mais dados do usuário.

Com isso vai dar para ter uma noção em média por onde tem a maior taxa de pessoas que entram na comunidade e assim quando necessário poder ter mais dados para talvez criar anúncio ou melhorar a abordagem de outras plataformas.

Algumas opções são:

- Twitch (Nossos streamers)
- Anúncios (quando tiverem anúncios ativos)
- Indicação (perguntar quem indicou talvez)
- Linkedin
- Instagram
- Twitter

**Comment by midhatdrops** [comment]
Algumas coisas importantes de salientar: Se for para captar esses dados para alguma função fim sem ser a de apresentação conforme dito, vale um termo de confirmação do compartilhamento de dados (LGPD means). Sobre perguntas obrigatórias, só deveriamos deixar obrigatório dados essenciais (nome, cidade, talvez idade). Por fim, como esses dados serão guardados ? Anonimizados? Como podemos garantir a segurança desses dados?
